rclcpp/include/rclcpp/publisher.hpp

```diff
@@ -132,6 +136,37 @@ class Publisher : public PublisherBase
     >::make_shared(size, this->get_allocator());
   }
 
+  /// Loan memory for a ROS message from the middleware
+  /**
+   * If the middleware is capable of loaning memory for a ROS message instance,
+   * the loaned message will be directly allocated in the middleware.
+   * If not, the message allocator of this rclcpp::Publisher instance is being used.
+   *
+   * With a call to \sa `publish` the LoanedMessage instance is being returned to the middleware
+   * or free'd accordingly to the allocator.
+   * If the message is not being published but processed differently, the message has to be
+   * returned by calling \sa `return_loaned_message`.
+   * \sa rclcpp::LoanedMessage for details of the LoanedMessage class.
+   *
+   * \return LoanedMessage containing memory for a ROS message of type MessageT
+   */
+  rclcpp::LoanedMessage<MessageT, AllocatorT>
```

Comment:
Should the AllocatorT be derived from the MessageT's allocator? Or is there a reason to be able to specify them separately I wonder?

Reply:
I guess when you do a vector of strings, you can specify the allocator for both separately, so maybe what's here already makes sense.